src/generate_article_json.py

```diff
@@ -24,19 +28,24 @@ def render(path):
         log = conf.multiprocess_log('generation.log', __name__)
         log.info(strbuffer.getvalue())
 
-def main():
-    paths = map(lambda fname: join(conf.XML_DIR, fname), os.listdir(conf.XML_DIR))
+def main(xml_dir, json_output_dir):
+    paths = map(lambda fname: join(xml_dir, fname), os.listdir(xml_dir))
     paths = filter(lambda path: path.lower().endswith('.xml'), paths)
     paths = sorted(paths, reverse=True)
-    #paths_sizes = zip(paths, map(os.path.getsize, paths))
-    #map(render, paths)
-    Parallel(n_jobs=-1)(delayed(render)(path) for path in paths)
+    Parallel(n_jobs=-1)(delayed(render)(path, json_output_dir) for path in paths)
     print 'see scrape.log for errors'
 
 if __name__ == '__main__':
-    '''
     import argparse
     parser = argparse.ArgumentParser()
-    args = parser.parse_args()
-    '''
-    main()
+    parser.add_argument('xml-dir', nargs='?', default=conf.XML_DIR)
+    parser.add_argument('output-dir', nargs='?', default=conf.JSON_DIR)
+
+    args = vars(parser.parse_args())
+    indir, outdir = [os.path.abspath(args[key]) for key in ['xml-dir', 'output-dir']]
+
+    ensure(os.path.exists(indir), "the path %r doesn't exist" % indir)
+    ensure(os.path.exists(outdir), "the path %r doesn't exist" % outdir)
+
+    main(indir, outdir)
+    exit(0)
```

Comment:
yeah. I saw that, thought the same thought and was too lazy to `ctrl-k` it. removing now.